gc.c

```diff
@@ -10178,31 +10204,31 @@ extern rb_symbols_t ruby_global_symbols;
 #define global_symbols ruby_global_symbols
 
 static void
-gc_update_references(rb_objspace_t *objspace)
+gc_update_references_heap(rb_objspace_t *objspace, rb_size_pool_t *size_pool, rb_heap_t *heap)
 {
-    rb_execution_context_t *ec = GET_EC();
-    rb_vm_t *vm = rb_ec_vm_ptr(ec);
-
     struct heap_page *page = NULL;
+    bool should_set_mark_bits = TRUE;
 
-    for (int i = 0; i < SIZE_POOL_COUNT; i++) {
-        bool should_set_mark_bits = TRUE;
-        rb_size_pool_t *size_pool = &size_pools[i];
-        rb_heap_t *heap = SIZE_POOL_EDEN_HEAP(size_pool);
-
-        list_for_each(&heap->pages, page, page_node) {
-            uintptr_t start = (uintptr_t)page->start;
-            uintptr_t end = start + (page->total_slots * size_pool->slot_size);
+    list_for_each(&heap->pages, page, page_node) {
+        uintptr_t start = (uintptr_t)page->start;
+        uintptr_t end = start + (page->total_slots * size_pool->slot_size);
 
-            gc_ref_update((void *)start, (void *)end, size_pool->slot_size, objspace, page);
-            if (page == heap->sweeping_page) {
-                should_set_mark_bits = FALSE;
-            }
-            if (should_set_mark_bits) {
-                gc_setup_mark_bits(page);
-            }
+        gc_ref_update((void *)start, (void *)end, size_pool->slot_size, objspace, page);
+        if (page == heap->sweeping_page) {
+            should_set_mark_bits = FALSE;
+        }
+        if (should_set_mark_bits) {
+            gc_setup_mark_bits(page);
         }
     }
+}
+
+static void
+gc_update_references_rest(rb_objspace_t *objspace)
```

Comment:
The name `gc_update_references_rest` is confusing. gc.c usese the `xyz_rest` name to mean that it will do `xyz` to completion (e.g. `gc_sweep_rest` finishes sweeping regardless of whether we're lazy sweeping or not).